codegen/src/types.rs

```diff
@@ -1356,3 +1390,64 @@ fn break_cycles(messages: &mut [Message], leaf_messages: &mut Vec<String>) {
 fn tag(number: u32, typ: &FieldType, packed: bool) -> u32 {
     number << 3 | typ.wire_type_num(packed)
 }
+
+/// "" is ("",""), "a" is ("","a"), "a.b" is ("a"."b"), and so forth.
+fn split_package(package: &str) -> (&str,&str) {
+    if package.is_empty() {
+        ("","")
+    } else {
+        if let Some(i) = package.rfind('.') {
+            (&package[0..i],&package[i+1..])
+        } else {
+            ("",package)
+        }
+    }
+}
+
+/// Given a file path, create or update the mod.rs file within its folder
+fn update_mod_file(path: &Path) -> Result<()> {
```

Comment:
This is still a concern for me:
- either the mod.rs has been built by the codegen: in this case **we know up-front all the necessary `pub mod` we need**
- or the mod.rs is an existing file: then we are potentially doing things wrong here!:
  - not only we are modifying existing files (dangerous!)
  - we are appending the new uses at the end of the file (ie after the code ...)